### PR TITLE
Include missing core object in glossary

### LINKED ISSUE #45972: Glossary list of core objects is wrong
**This is a Bug Report**

**Problem:**

The glossary listing for [core objects](https://k8s.io/docs/reference/glossary/?core-object=true) includes several things that aren't core object kinds, such as:
- Pod Security Policy (removed!)
- Quantity (not an API kind that you can create using `kubectl` etc)
- Dynamic Volume Provisioning (not an API kind at all)
- Cloud Controller Manager (a binary, not an API kind)
- Taint (a fundamental concept, but not an API kind)
- Volume (not an API kind that you can create using `kubectl` etc)
- Volume Plugin (not an API kind at all)
- QoS Class (not an API kind)

**Proposed Solution:**

Edit those glossary entries. Optionally add in glossary entries for all core object kinds; for example, NetworkPolicy, Node and PodTemplate should be in the list.

**Page to Update:**
https://k8s.io/docs/reference/glossary/?core-object=true

**Additional Information:**
/language en

### PR DESCRIPTION
**Pull request by abhilasha2418**
### Description
* This is the first part of resolving the issue regarding core objects in the Kubernetes glossary.
*  In this PR, I have included necessary API kinds that are built into Kubernetes as core objects, such as Node, PodTemplate, and NetworkPolicy.

### Issue

#45972

### CODE REVIEW

File: content/en/docs/reference/glossary/pod-template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Pod Template
```
Comment (sftim) [inline content/en/docs/reference/glossary/pod-template.md:2]:
```suggestion
title: PodTemplate
```

File: content/en/docs/reference/glossary/pod-template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+id: pod-template
+date: 2024-10-13
+short_description: >
+  A Pod Template defines template for creating pods.
```
Comment (sftim) [inline content/en/docs/reference/glossary/pod-template.md:6]:
```suggestion
  A template for creating Pods.
```

File: content/en/docs/reference/glossary/pod-template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+aka:
```
Comment (sftim) [inline content/en/docs/reference/glossary/pod-template.md:8]:
```suggestion
aka:
  - pod template
```

File: content/en/docs/reference/glossary/pod-template.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+date: 2024-10-13
+short_description: >
+  A Pod Template defines template for creating pods.
+
+aka: 
+tags:
+- core-object
+
+---
+An API object that defines a template for creating {{< glossary_tooltip text="pod" term_id="pod" >}}, typically used in higher-level controllers.
```
Comment (sftim) [inline content/en/docs/reference/glossary/pod-template.md:13]:
```suggestion
An API object that defines a template for creating {{< glossary_tooltip text="pod" term_id="pod" >}}.
The PodTemplate API is also embedded in API definitions for workload management, such as 
{{< glossary_tooltip text="Deployment" term_id="deployment" >}} or
{{< glossary_tooltip text="StatefulSets" term_id="StatefulSet" >}}.
```

File: content/en/docs/reference/glossary/pod-template.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+<!--more--> 
+
+Pod Templates provide a specification that includes metadata, labels, and a pod's desired state, which is utilized by controllers like {{< glossary_tooltip text="deployment" term_id="deployment" >}} and {{< glossary_tooltip text="statefulset" term_id="statefulset" >}} to create and manage multiple {{< glossary_tooltip text="replicas" term_id="" >}} of {{< glossary_tooltip text="pod" term_id="pod" >}}.
```
Comment (sftim) [inline content/en/docs/reference/glossary/pod-template.md:17]: Try this:
```suggestion
Pod templates allow you to define common metadata (such as labels, or a template for the name of a
new Pod) as well as to specify a pod's desired state.
[Workload management](/docs/concepts/workloads/controllers/) controllers use Pod templates
to define and manage one or more {{< glossary_tooltip text="Pods" term_id="pod" >}}.
When there can be multiple Pods based on the same template, these are called
{{< glossary_tooltip term_id="replica" text="replicas" >}}.

Although you can create a PodTemplate object directly, you rarely need to do so.
```

**Review by sftim — COMMENTED** [review]
The core object is called PodTemplate; let's fix that.

**Comment by abhilasha2418** [comment]
@sftim I've added the changes as you suggested!

**Comment by sftim** [comment]
I'm going to edit the issue description to remove

> Closes https://github.com/kubernetes/website/issues/45972

as this PR is not a complete fix.

**Comment by abhilasha2418** [comment]
@dipesh-rawat can you please approve the PR.

**Review by dipesh-rawat — COMMENTED** [review]
Thanks!
/approve

**Pull request merged**